### PR TITLE
In query-frontend, log details of each request to OpenTracing

### PR DESCRIPTION
**Pull request by bboreham**
This is to help with debugging: the top-level request that comes in and what is sent to queriers are logged to the current span, so you can see the effects of splitting, caching, etc.

### CODE REVIEW

File: pkg/querier/frontend/frontend.go [hunk 1]
```diff
@@ -316,6 +316,9 @@ func (f *Frontend) Process(server Frontend_ProcessServer) error {
 		}
 
 		originalCtx := request.originalCtx
+		if request.request != nil && request.request.QueryRangeRequest != nil {
```
Comment (tomwilkie) [inline pkg/querier/frontend/frontend.go]: Under what circumstances will request.request be nil?  I can't think of any.

Comment (bboreham) [reply pkg/querier/frontend/frontend.go]: Idk it crashed in the unit tests and I couldn’t be bothered figuring out which one was nil.

Comment (tomwilkie) [reply pkg/querier/frontend/frontend.go]: I removed `request.request != nil &&` and the tests passed...

Either way this is in the wrong place; if you look at L172, `QueryRangeRequest` is never initialised in `ProcessRequest`.  I don't even recall why its there.  We should probably remove it, and put this call into `queryRangeTerminator`.

Comment (bboreham) [reply pkg/querier/frontend/frontend.go]: Thanks for the tip!

**Comment by bboreham** [comment]
Looks like this:
image

File: pkg/querier/frontend/query_range.go [hunk 2]
```diff
@@ -91,6 +92,15 @@ func (q QueryRangeRequest) toHTTPRequest(ctx context.Context) (*http.Request, er
 	return req.WithContext(ctx), nil
 }
 
+func (q QueryRangeRequest) logToSpan(ctx context.Context) {
+	if span := opentracing.SpanFromContext(ctx); span != nil {
+		span.LogFields(otlog.String("query", q.Query))
+		span.LogFields(otlog.String("start", timestamp.Time(q.Start).String()))
+		span.LogFields(otlog.String("end", timestamp.Time(q.End).String()))
+		span.LogFields(otlog.Int64("step (ms)", q.Step))
```
Comment (tomwilkie) [inline pkg/querier/frontend/query_range.go]: Nit: should this be one call to LogFields (its variadic)?  Not sure what the best practices here are.

**Comment by tomwilkie** [comment]
LGTM with one nit.

**Pull request merged**